plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3RepositoryPlugin.java

```diff
@@ -387,8 +389,14 @@ public void reload(Settings settings) {
     public void close() throws IOException {
         service.close();
         s3AsyncService.close();
-        urgentExecutorBuilder.getAsyncTransferEventLoopGroup().close();
-        priorityExecutorBuilder.getAsyncTransferEventLoopGroup().close();
-        normalExecutorBuilder.getAsyncTransferEventLoopGroup().close();
+        if (urgentExecutorBuilder.getAsyncTransferEventLoopGroup() != null) {
+            urgentExecutorBuilder.getAsyncTransferEventLoopGroup().close();
+        }
+        if (priorityExecutorBuilder.getAsyncTransferEventLoopGroup() != null) {
+            priorityExecutorBuilder.getAsyncTransferEventLoopGroup().close();
+        }
+        if (normalExecutorBuilder.getAsyncTransferEventLoopGroup() != null) {
+            normalExecutorBuilder.getAsyncTransferEventLoopGroup().close();
+        }
```

Comment:
this code can be written here - https://github.com/awslabs/aws-crt-java/blob/f4e09df6e3bb14c40c6a80cf474d2afd170de32f/src/main/java/software/amazon/awssdk/crt/CrtResource.java#L427